### PR TITLE
EGM ID Update 76X

### PR DESCRIPTION
**Pull request by lgray**
Forward port of #12738.

New photon ID added to 76X MiniAOD.

### CODE REVIEW

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by lgray** [comment]
@ikrav

File: PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py [hunk 1]
```diff
@@ -185,7 +185,7 @@ def miniAOD_customizeCommon(process):
         setupAllVIDIdsInModule(process,idmod,setupVIDElectronSelection,None,False)
 
     #VID Photon IDs
-    photon_ids = ['RecoEgamma.PhotonIdentification.Identification.cutBasedPhotonID_PHYS14_PU20bx25_V2p1_cff',
+    photon_ids = ['RecoEgamma.PhotonIdentification.Identification.cutBasedPhotonID_Spring15_25ns_V1_cff',
```
Comment (slava77) [inline PhysicsTools/PatAlgos/python/slimming/miniAOD_tools.py]: will this affect the ongoing (re)miniAODv2 validation?

**Comment by cvuosalo** [comment]
+1

For #12742 68e338145ad09e1589c0bd1a0438e1d268cc9f0f

Adding a new photon ID. #12740 and #12738 are the 80X and 74X versions of this PR, and they both have already been approved by Reco.

The code changes are satisfactory, and Jenkins tests against baseline CMSSW_7_6_X_2015-12-09-2300 show no significant differences. An extended test performed for #12740 also shows no significant differences.

This approval assumes this PR will not interfere with the ongoing Mini-AOD validation. If this assumption is not correct, then this PR should not go forward at this time.

**Comment by lgray** [comment]
@monttj @vadler Please review this PR.

**Comment by lgray** [comment]
@davidlange6 When will this be merged? I suspect you are waiting for a prerelease to work for 80X? It is odd that this is in 74X/80X and not 76X

**Comment by slava77** [comment]
@lgray 
this PR modifies miniAOD.
so, it will not be merged to 76X until there is a (re)miniAODv3 (if ever).

**Comment by lgray** [comment]
Ah, that would have been nice to know when Carl +1'd this two weeks ago (there was some text, but no response at all...). It is easy to make this PR *not* change MiniAOD, and as usual the main point it getting the IDs in the release rather than poking MiniAOD.

If I remove the commit can this be +1'd quickly and put into 623_patchX?

Thanks.

**Comment by slava77** [comment]
back when it was +1 there was still a possibility for release used for data rereco and miniAODv2.
In the end, there was no release after 76X and the new processing campaign is ongoing since ~Dec 17.

You can update this PR to not modify miniAOD.
My guess is that a new 76X will be made in early January if needed.

**Comment by lgray** [comment]
@cmsbuild please test

**Comment by lgray** [comment]
Ohh neat, I appear to have confused cms-bot state machine and the reco-signed flag wasn't removed! (and therefore this PR became fully signed)

**Comment by slava77** [comment]
now you know how to get some large code changes in
without disturbing signatures

**Comment by lgray** [comment]
@smuzaffar You may want to address this bug in cms-bot.

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/12742	00e4da2
- compared to the previously signed version, there is no modification to the miniAOD outputs and the code is essentially the same as in the merged #12738 equivalent in 74X
- jenkins tests pass and comparisons show no differences (sanity check mostly)

**Comment by davidlange6** [comment]
@slava77 , @lgray - is there some documentation/presentation of these changes that we can refer to in the PR for completeness?

**Comment by lgray** [comment]
It's all on the EGM TWiki. Please merge this PR.

**Comment by slava77** [comment]
Lindsey, please add the link to the twiki in the PR description.

**Comment by davidlange6** [comment]
+1

indeed, that was the idea - I think its this:
https://twiki.cern.ch/twiki/bin/view/CMS/CutBasedPhotonIdentificationRun2

**Comment by lgray** [comment]
Thanks for merging this. Yes that is the right link. I'll be sure to include additional supporting material next time this goes in the 80X PR so it is linked to the backports. This is the first time more material was requested for an EGM ID update.

**Pull request merged**